src/libraries/System.Text.RegularExpressions/gen/UpgradeToGeneratedRegexCodeFixer.cs

```diff
@@ -287,6 +287,20 @@ static RegexOptions GetRegexOptionsFromArgument(ImmutableArray<IArgumentOperatio
                 {
                     return literalOperation.Syntax;
                 }
+                else if (argument.Value is IFieldReferenceOperation fieldReferenceOperation &&
+                    fieldReferenceOperation.Member is IFieldSymbol fieldSymbol && fieldSymbol.IsConst)
+                {
+                    return generator.Argument(fieldReferenceOperation.Syntax);
+                }
+                else if (argument.Value.ConstantValue.HasValue && argument.Value.ConstantValue.Value!.ToString()!.Contains('\\'))
```

Comment:
>  argument.Value.ConstantValue.Value!.ToString()!.Contains('\\')

As it is converting to string anyways I think using string interpolation will be at least more readable
```cs
    else if (argument.Value.ConstantValue.Value is string str && str.Contains('\\'))
    {
        return SyntaxFactory.ParseExpression($"@\"{str}\"");
    }
```